### PR TITLE
Testing scripts

### PR DESCRIPTION
**Pull request by aksheeth-i**
Introducing automated scripts to test iDFlakies' functionality on various maven projects/configurations on CI.

### CODE REVIEW

File: .github/workflows/test-many-configs.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# This is a basic workflow to help you get started with Actions
```
Comment (august782) [inline .github/workflows/test-many-configs.yml:1]: Probably want to strip out these comments leftover from the initial tutoria/template.

File: .github/workflows/test-many-configs.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+name: auto-run-manyConfigs-script
```
Comment (august782) [inline .github/workflows/test-many-configs.yml:3]: I think this should change to "auto-run-test-many-configs-script" to match the name.

File: .github/workflows/test-many-configs.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+# Controls when the workflow will run
+on: [push]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  manyConfigsTest1:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    # timeout-minutes: 4320
```
Comment (august782) [inline .github/workflows/test-many-configs.yml:14]: Do we need a timeout for this one?

File: .github/workflows/test-many-projects.yml [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+      - name: Set up JDK 8
+        uses: actions/setup-java@v2
+        with:
+          java-version: '8'
+          distribution: 'adopt'
+
+      - name: Installs packages
+        run: mvn install -B
+
+      - name: Test 3rd set of projects
```
Comment (august782) [inline .github/workflows/test-many-projects.yml:121]: 3rd -> 4th

File: scripts/idfFailures.csv [hunk 5]
```diff
@@ -0,0 +1,2 @@
+https://github.com/wildfly/wildfly,b19048b72669fc0e96665b1b125dc1fda21f5993,naming,0
```
Comment (august782) [inline scripts/idfFailures.csv:1]: What is this file used for again?

File: scripts/idfProjects.csv [hunk 6]
```diff
@@ -0,0 +1 @@
+https://github.com/wildfly/wildfly,b19048b72669fc0e96665b1b125dc1fda21f5993,naming
```
Comment (august782) [inline scripts/idfProjects.csv:1]: Is this file needed?

File: scripts/test-many-projects.sh [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+    bash ./modify-project.sh ${projectDirectory} 1.2.0-SNAPSHOT
+    cd ${projectDirectory}
+
+
+
+
+    function checkFlakyTests() {
+        expectedTests=$1
+        if [[ ${expectedTests} == -1 ]]; then
+            echo "EXPECTED PROJECT FAILURE %%%%%"
```
Comment (august782) [inline scripts/test-many-projects.sh:69]: Does it not return anything here? Are there now any expected failures?

File: scripts/test-many-configs.sh [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+
+if [[ ! -d MC-script-results ]]; then
+    mkdir MC-script-results
+fi
+cd MC-script-results
+
+
+
+function cleanUp() {
+    if [[ ${2} != "" ]]; then
```
Comment (august782) [inline scripts/test-many-configs.sh:30]: Give these arguments some local variable names to make it easier to understand what they are supposed to represent.

File: scripts/test-many-configs.sh [hunk 9]
```diff
@@ -0,0 +1,345 @@
...
+
+
+
+
+#6. Save all relevant data
+
+if [[ ! -d ARTIFACTS ]]; then
+    mkdir ARTIFACTS
+fi
+for d in test{1,2,3,4,5,6,7,8,9}; do
```
Comment (august782) [inline scripts/test-many-configs.sh:335]: For these test* directories, are they saved right there in the current directory where you are at (which I believe is MC-script-results/)? The following logic seems to be checking that the directory, e.g., test1/, is directly in the current directory, but then to copy it the script tries to $(find -name ${d}), which suggests it is somehow nested underneath somewhere?

File: scripts/test-many-projects.sh [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+        flag=1
+    else
+
+        #5. Run the default test for each
+        mvn testrunner:testplugin -Ddetector.detector_type=random-class-method -Ddt.randomize.rounds=5 -Ddt.detector.original_order.all_must_pass=false -Ddt.detector.roundsemantics.total=true ${MVNOPTIONS} ${PL} -B
+        if [[ $? != 0 ]]; then
+            echo "${URL} idflakies detect not successful. %%%%%"
+            flag=1
+        else
+           checkFlakyTests ${numTests} ${URL} $(echo ${MODULE} | cut -d'|' -f1)
```
[the remark this answers is not in the document]
Comment (august782) [reply scripts/test-many-projects.sh:141]: Also, I think you should include some comment about that logic pertaining to "|" in the MODULE here.

File: .github/workflows/test-many-configs.yml [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+jobs:
+  manyConfigsTest:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up JDK 8
+        uses: actions/setup-java@v2
+        with:
+          java-version: '8'
```
Comment (august782) [inline .github/workflows/test-many-configs.yml:16]: Would be nice if we could configure exactly which version of Java 8, if that is even possible with this Action setup.

File: .github/workflows/test-many-configs.yml [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+          distribution: 'adopt'
+
+      - name: Installs packages
+        run: mvn install -B
+
+      - name: Test set of projects
+        run: set -o pipefail ; bash scripts/test-many-configs.sh scripts/testExpectations.csv |& tee -a scripts/GeneralLogs.log
+
+      - name: Move Logs file into the Artifact folder
```
Comment (august782) [inline .github/workflows/test-many-configs.yml:25]: Artifact -> ARTIFACTS

File: .github/workflows/test-many-configs.yml [hunk 13]
```diff
@@ -0,0 +1,35 @@
+
+name: auto-run-test-many-configs-script
+
+on: [push]
+
+
```
Comment (august782) [inline .github/workflows/test-many-configs.yml:6]: Delete extra newline.

File: .github/workflows/test-many-projects.yml [hunk 14]
```diff
@@ -0,0 +1,119 @@
+
+name: auto-run-test-many-projects-script
+
+on: [push]
+
+jobs:
+  manyProjTest1:
+    runs-on: ubuntu-latest
+    steps:
```
Comment (august782) [inline .github/workflows/test-many-projects.yml:9]: Check if it's possible to have some way to describe the same steps across all jobs, since all of them use the same setup.

File: .github/workflows/test-many-projects.yml [hunk 15]
```diff
@@ -0,0 +1,119 @@
...
+          java-version: '8'
+          distribution: 'adopt'
+
+      - name: Installs packages
+        run: mvn install -B
+
+      - name: Test 1st set of projects
+        run: set -o pipefail ; bash scripts/test-many-projects.sh scripts/idfProjects1.csv |& tee -a scripts/GeneralLogs.log
+
+      - name: Move Logs file into the Artifact folder
```
Comment (august782) [inline .github/workflows/test-many-projects.yml:24]: Artifact -> ARTIFACTS

File: .github/workflows/test-many-configs.yml [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Move Logs file into the Artifact folder
+        if: always()
+        run: mv scripts/GeneralLogs.log scripts/MC-script-results/ARTIFACTS/
+
+      - name: Save all relevant files
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+         name: artifact_1
```
Comment (august782) [inline .github/workflows/test-many-configs.yml:33]: Rename artifact_1 to represent that it is output of the many configs script.

File: .github/workflows/test-many-projects.yml [hunk 17]
```diff
@@ -0,0 +1,119 @@
...
+        if: always()
+        run: mv scripts/GeneralLogs.log scripts/testing-script-results/ARTIFACTS/
+
+      - name: Save all relevant files
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+         name: artifact_1
```
Comment (august782) [inline .github/workflows/test-many-projects.yml:32]: Change name of artifacts to match what script it came from.

File: .github/workflows/test-many-projects.yml [hunk 18]
```diff
@@ -0,0 +1,119 @@
...
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+         name: artifact_2
+         path: scripts/testing-script-results/ARTIFACTS
+
+  manyProjTest3:
+    runs-on: ubuntu-latest
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
```
Comment (august782) [inline .github/workflows/test-many-projects.yml:66]: No need for this comment.

File: scripts/idfFailures.txt [hunk 19]
```diff
@@ -0,0 +1 @@
+1) https://github.com/apache/incubator-dubbo,737f7a7ea67832d7f17517326fb2491d0a086dd7,dubbo-registry/dubbo-registry-default|dubbo-dependencies-bom         - this project had a problem with ____
```
Comment (august782) [inline scripts/idfFailures.txt:1]: Is this missing some notes?

File: scripts/idfProjects1.csv [hunk 20]
```diff
@@ -0,0 +1,9 @@
+https://github.com/wildfly/wildfly,b19048b72669fc0e96665b1b125dc1fda21f5993,naming,0
```
Comment (august782) [inline scripts/idfProjects1.csv:1]: Would be nice to have a header to describe each column, e.g., "#project,SHA,module,expectedFlakyTests"

File: scripts/test-many-configs.sh [hunk 21]
```diff
@@ -0,0 +1,419 @@
...
+
+if [[ ${1} == "" ]]; then
+    echo "Please provide the name of your csv file with the format "URL,SHA,MODULE,tests1-9" on each line."
+    exit
+fi
+
+
+#0. File management/defining all helper functions
+
+flag=0  #Global flag designed to represent the status of the overall status of the whole build when running on a CI
```
Comment (august782) [inline scripts/test-many-configs.sh:12]: "Global flag designed to represent the overall status of the whole build when running on CI"

File: scripts/test-many-configs.sh [hunk 22]
```diff
@@ -0,0 +1,419 @@
+#!/bin/bash
```
Comment (august782) [inline scripts/test-many-configs.sh:1]: Should we put a note that this script should only ever be run on the GitHub Actions CI, so one should look at the corresponding yml files to see how to use this script?

File: scripts/test-many-configs.sh [hunk 23]
```diff
@@ -0,0 +1,419 @@
...
+scriptDir=$(cd $(dirname $0); pwd)
+csvFile=$(cd $(dirname $1); pwd)/$(basename $1)
+cd ${scriptDir}
+
+if [[ ! -f ${csvFile} ]]; then
+    echo "No such csv exists in the given directory."
+    exit
+fi
+
+if [[ ! -d MC-script-results ]]; then
```
Comment (august782) [inline scripts/test-many-configs.sh:22]: Add some quick comment that MC-script-results/ is where all the results will go at the end.

File: scripts/test-many-configs.sh [hunk 24]
```diff
@@ -0,0 +1,419 @@
...
+    mkdir MC-script-results
+fi
+cd MC-script-results
+
+
+function setOriginalOrder() {
```
Comment (august782) [inline scripts/test-many-configs.sh:28]: Add some quick comment that this function copies the original order of tests we want to use in, as to avoid recomputing it each time (which can be different).

File: scripts/test-many-configs.sh [hunk 25]
```diff
@@ -0,0 +1,419 @@
...
+    testNum=$1
+    currModule=$2
+    if [[ ${currModule} != "" ]]; then
+        cd ${currModule}
+    fi
+    if [[ ! -d ".dtfixingtools" ]]; then
+        cd ${projectDirectory}
+        return 0
+    fi
+    cp -r .dtfixingtools ${testNum}
```
Comment (august782) [inline scripts/test-many-configs.sh:55]: You could also just "mv" the directory.

File: scripts/test-many-configs.sh [hunk 26]
```diff
@@ -0,0 +1,419 @@
...
+    if [[ ${currModule} != "" ]]; then
+        cd ${currModule}
+    fi
+    if [[ ! -d ${testNum} ]]; then
+        cd ${projectDirectory}
+        return 0
+    fi
+    cd ${testNum}
+    cd detection-results
+    if [[  -d ${detectType} ]]; then
```
Comment (august782) [inline scripts/test-many-configs.sh:75]: Remove extra spacing.

File: scripts/test-many-configs.sh [hunk 27]
```diff
@@ -0,0 +1,419 @@
...
+        cp ${scriptDir}/original-order-files/${projName} .dtfixingtools/original-order
+    else
+        cp ${scriptDir}/original-order-files/${currModule} .dtfixingtools/original-order
+    fi
+    cd ${projectDirectory}
+}
+
+
+
+function cleanUp() {
```
Comment (august782) [inline scripts/test-many-configs.sh:45]: For each function, would be nice to have a quick, one-line comment of sorts to describe the point of each function.

File: scripts/test-many-configs.sh [hunk 28]
```diff
@@ -0,0 +1,419 @@
...
+    if [[ ${currModule} != "" ]]; then
+        cd ${currModule}
+    fi
+    if [[ ! -d ${testNum} ]]; then
+        cd ${projectDirectory}
+        return 0
+    fi
+    cd ${testNum}
+    cd detection-results
+    if [[  -d ${detectType} ]]; then
```
Comment (august782) [inline scripts/test-many-configs.sh:102]: Remove extra space.

File: scripts/test-many-configs.sh [hunk 29]
```diff
@@ -0,0 +1,419 @@
...
+        cd ${detectType}
+        numRounds=$(ls | wc -l)
```
Comment (august782) [inline scripts/test-many-configs.sh:104]: Comment that the number of rounds run is the same as the number of files here.

File: scripts/test-many-configs.sh [hunk 30]
```diff
@@ -0,0 +1,419 @@
...
+        if [ ${numRounds} ${expectedBehavior} ${expectedRounds} ]; then
```
Comment (august782) [inline scripts/test-many-configs.sh:105]: The variable expectedBehavior is a bit odd in how we should expect it to be used, warrants some comment about what it should be.

File: scripts/test-many-configs.sh [hunk 31]
```diff
@@ -0,0 +1,419 @@
...
+            echo "Success: ${testNum} ran ${numRounds} round(s) as expected."
+            cd ${projectDirectory}
+            return 0
+        else
+            echo "Error: ${testNum} ran ${numRounds} round(s) when it should have ran ${expectedRounds}. %%%%%"
```
Comment (august782) [inline scripts/test-many-configs.sh:110]: Maybe want to add the ${expectedBehavior} in, so we can see the expected relationship between numbers here, e.g., < or >.

File: scripts/test-many-configs.sh [hunk 32]
```diff
@@ -0,0 +1,419 @@
...
+        return 0
+    fi
+    cd ${testNum}
+    cd detection-results
+    if [[  -d ${detectType} ]]; then
+        echo "Success: Flaky tests were found using the ${detectType} detector type."
+        cd ${projectDirectory}
+        return 0
+    fi
+    echo "No flaky tests were found using the ${detectType} detector type."
```
Comment (august782) [inline scripts/test-many-configs.sh:80]: Maybe the issue should be that the detector was not run (since a detector still generates that directory even when no flaky tests detected); would then be a failure when this happens.

File: scripts/test-many-configs.sh [hunk 33]
```diff
@@ -0,0 +1,419 @@
...
+            return 0
+        else
+            echo "Error: ${testNum} ran ${numRounds} round(s) when it should have ran ${expectedRounds}. %%%%%"
+            flag=1
+            cd ${projectDirectory}
+            return 1
+        fi
+    fi
+    cd ${projectDirectory}
+    echo "No flaky tests were found using ${expectedRounds} rounds."
```
Comment (august782) [inline scripts/test-many-configs.sh:117]: Remove this message, does not match what this function is trying to do.

File: scripts/test-many-configs.sh [hunk 34]
```diff
@@ -0,0 +1,419 @@
...
+}
+
+
+
+function checkAbsPath() {
+    path=$1
+    resultsFolder=$2
```
Comment (august782) [inline scripts/test-many-configs.sh:124]: resultsFolder -> resultsDirectory

File: scripts/test-many-configs.sh [hunk 35]
```diff
@@ -0,0 +1,419 @@
...
+    return 1
+}
+
+
+
+function flakyTestsFound() {
+    testNum=$1
+    expectedTests=$2
+    currModule=$3
+    if [[ $expectedTests == -* ]]; then
```
Comment (august782) [inline scripts/test-many-configs.sh:151]: Put a quick comment about how having a "-" as a prefix for this entity means expecting some error message.

File: scripts/test-many-configs.sh [hunk 36]
```diff
@@ -0,0 +1,419 @@
...
+        else
+            echo "ERROR: Unexpected Project Failure in ${testNum} %%%%%"
+            flag=1
+            return 1
+        fi
+    else
+        if [[ ${currModule} != "" ]]; then
+            cd ${currModule}
+        fi
+        cd ${testNum}
```
Comment (august782) [inline scripts/test-many-configs.sh:165]: Put a quick comment that results from the run were all stored into ${testNum}.

File: scripts/test-many-configs.sh [hunk 37]
```diff
@@ -0,0 +1,419 @@
...
+        cd detection-results
+        numFlakyTests=$(awk '1' list.txt | wc -l)
```
Comment (august782) [inline scripts/test-many-configs.sh:167]: Quick comment that this simply counts number of lines in the list.txt file.

File: scripts/test-many-configs.sh [hunk 38]
```diff
@@ -0,0 +1,419 @@
...
+        fi
+    fi
+}
+
+
+
+function checkTimeout() {
+    testNum=$1
+    expectedTime=$2
+    if [[ ! -f ${testNum} ]]; then
```
Comment (august782) [inline scripts/test-many-configs.sh:194]: ${testNum} -> ${testsNum}.log

File: scripts/test-many-configs.sh [hunk 39]
```diff
@@ -0,0 +1,419 @@
...
+        return 0
+    fi
+
+    if grep "Using a timeout of ${expectedTime}" ${testNum}.log; then
+        echo "Success: ${testNum} used a timeout of ${expectedTime} as expected."
+        return 0
+    else
+        echo "Error: ${testNum} did not use a timeout. %%%%%"
```
Comment (august782) [inline scripts/test-many-configs.sh:202]: Technically, message should be that it did not use the expected timeout.

File: scripts/test-many-configs.sh [hunk 40]
```diff
@@ -0,0 +1,419 @@
...
+
+
+function cleanUp() {
+    testNum=$1
+    currModule=$2
+    if [[ ${currModule} != "" ]]; then
+        cd ${currModule}
+    fi
+    if [[ ! -d ".dtfixingtools" ]]; then
+        cd ${projectDirectory}
```
Comment (august782) [inline scripts/test-many-configs.sh:52]: Maybe best to in general make a new variable name, e.g., currDir, at the top of each function, where "currDir=$(pwd)", as to ensure we know each function starts and ends in the same place.

File: scripts/test-many-configs.sh [hunk 41]
```diff
@@ -0,0 +1,419 @@
...
+
+    #4 Maven install the proj
+
+    if [[ ${MODULE} != "" ]]; then
+        PL="-pl ${MODULE}"
+    else
+        PL=""
+    fi
+
+    if [[ $URL == "https://github.com/wildfly/wildfly" ]]; then
```
Comment (august782) [inline scripts/test-many-configs.sh:252]: Put a comment about how it is necessary to modify the repository location on this old version of wildfly.

File: scripts/test-many-configs.sh [hunk 42]
```diff
@@ -0,0 +1,419 @@
...
+        semantics="-Ddt.detector.roundsemantics.total="
+        absPath="-Ddt.cache.absolute.path="
+
+
+
+        setOriginalOrder ${starr[4]} ${MODULE}
+        #Test 1: Simply test if the timeout configuration is working. Not testing accuracy here since it has proven to be nondeterministic
+        set -o pipefail ; mvn testrunner:testplugin ${time}120 ${ogOrderPass}true ${detType}random-class-method ${PL} |& tee -a ${projectDirectory}/test1.log
+        if [[ $? != 0 ]]; then
+            echo "${URL} testrunner was not successful. %%%%%"
```
Comment (august782) [inline scripts/test-many-configs.sh:286]: testrunner -> iDFlakies

File: scripts/test-many-configs.sh [hunk 43]
```diff
@@ -0,0 +1,419 @@
...
+            flag=1
+        fi
+        cleanUp test5 ${MODULE}
+        checkNumberRounds test5 8 random-class-method -ge ${MODULE}
+        flakyTestsFound test5 "$testCount5" ${MODULE}
+
+
+        setOriginalOrder ${starr[4]} ${MODULE}
+        #Test 6: Try the reverse determinant type. Make sure only 1 round of tests is ran despite rounds being specified
```
Comment (august782) [inline scripts/test-many-configs.sh:348]: is ran -> is run

File: scripts/test-many-configs.sh [hunk 44]
```diff
@@ -0,0 +1,419 @@
...
+            echo "${URL} testrunner was not successful. %%%%%"
+            flag=1
+        fi
+        cleanUp test7 ${MODULE}
+        checkNumberRounds test7 1 reverse-class -ge ${MODULE}
+        flakyTestsFound test7 "$testCount7" ${MODULE}
+
+
+        setOriginalOrder ${starr[4]} ${MODULE}
+        #Test 8: Try setting semantics to be true, where there should be exactly as many rounds as specified. No more, no less
```
Comment (august782) [inline scripts/test-many-configs.sh:372]: semantics -> round semantics

File: scripts/test-many-configs.sh [hunk 45]
```diff
@@ -0,0 +1,419 @@
...
+        echo "Error: ${testNum} did not use a timeout. %%%%%"
+        flag=1
+        return 1
+    fi
+}
+
+
+
+
+while IFS="," read -r URL SHA MODULE testCount1 testCount2 testCount3 testCount4 testCount5 testCount6 testCount7 testCount8 testCount9; do
```
Comment (august782) [inline scripts/test-many-configs.sh:211]: Note, if we introduce a header that starts with # in the input file, would need to skip that first line.

File: scripts/test-many-projects.sh [hunk 46]
```diff
@@ -0,0 +1,161 @@
...
+    if [[ ${expectedTests} == -1 ]]; then
+        echo "EXPECTED PROJECT FAILURE %%%%%"
+        return 0
+    else
+        if [[ ${currModule} != "" ]]; then
+            cd ${currModule}
+        fi
+        cd .dtfixingtools
+        cd detection-results
+        numFlakyTests=$(awk '1' list.txt | wc -l)
```
Comment (august782) [inline scripts/test-many-projects.sh:40]: Same as before, comment about this is counting.

File: scripts/test-many-projects.sh [hunk 47]
```diff
@@ -0,0 +1,161 @@
...
+    projectURL=$2
+    currModule=$3
+    if [[ ${expectedTests} == -1 ]]; then
+        echo "EXPECTED PROJECT FAILURE %%%%%"
+        return 0
+    else
+        if [[ ${currModule} != "" ]]; then
+            cd ${currModule}
+        fi
+        cd .dtfixingtools
```
Comment (august782) [inline scripts/test-many-projects.sh:38]: Could simplify to .dtfixingtools/detection-results/

File: scripts/test-many-projects.sh [hunk 48]
```diff
@@ -0,0 +1,161 @@
...
+
+
+
+#CSV Splicing:
+while IFS="," read -r URL SHA MODULE numTests; do
+
+    renamedRepo=${URL}"/"
+    readarray -d / -t starr <<< "${renamedRepo}"
+
+    echo "DATE: "
```
Comment (august782) [inline scripts/test-many-projects.sh:81]: Do you want to put the same date command in the other script?

File: scripts/test-many-projects.sh [hunk 49]
```diff
@@ -0,0 +1,161 @@
...
+    bash ${scriptDir}/../pom-modify/modify-project.sh ${projectDirectory} 1.2.0-SNAPSHOT
+
+
+
+
+
+    #4. Maven install the proj
+
+    if [[ ${MODULE} != "" ]]; then
+        PL="-pl $(echo ${MODULE} | sed 's;|;,;g')"  #some projects rely on multiple dependendencies to be installed, so the | delimited we used in csv must be converted into a comma for mvn install
```
Comment (august782) [inline scripts/test-many-projects.sh:115]: delimited -> delimiter

File: scripts/test-many-projects.sh [hunk 50]
```diff
@@ -0,0 +1,161 @@
...
+    else
+        PL=""
+    fi
+    MVNOPTIONS="-Dfindbugs.skip=true -Dmaven.javadoc.skip=true -Denforcer.skip=true -Drat.skip=true -Dmdep.analyze.skip=true -Dgpg.skip -Dmaven.javadoc.skip=true -Ddependency-check.skip=true"
```
Comment (august782) [inline scripts/test-many-projects.sh:119]: Might be good to have these same skip options for the other script.

File: scripts/test-many-configs.sh [hunk 51]
```diff
@@ -0,0 +1,419 @@
...
+        countFirstFail="-Ddt.detector.count.only.first.failure="
+        verRounds="-Ddt.verify.rounds="
+        time="-Ddetector.timeout="
+        semantics="-Ddt.detector.roundsemantics.total="
+        absPath="-Ddt.cache.absolute.path="
+
+
+
+        setOriginalOrder ${starr[4]} ${MODULE}
+        #Test 1: Simply test if the timeout configuration is working. Not testing accuracy here since it has proven to be nondeterministic
```
Comment (august782) [inline scripts/test-many-configs.sh:283]: Propagate the same options for running testrunner from the other script to relevant places in this script.

File: scripts/testExpectations.csv [hunk 52]
```diff
@@ -0,0 +1,6 @@
+https://github.com/wildfly/wildfly,b19048b72669fc0e96665b1b125dc1fda21f5993,naming,32,0,44,0,0,0,0,31,33
```
Comment (august782) [inline scripts/testExpectations.csv:1]: Same comment about potentially having some header.

File: .github/workflows/test-many-configs.yml (hunk 11 above)
Comment (aksheeth-i) [reply .github/workflows/test-many-configs.yml:16]: From GH Actions documentation:
"The java-version input supports an exact version or a version range using SemVer notation:

major versions: 8, 11, 16, 17
more specific versions: 17.0, 11.0, 11.0.4, 8.0.232, 8.0.282+8
early access (EA) versions: 15-ea, 15.0.0-ea, 15.0.0-ea.2, 15.0.0+2-ea"

Out of the more specific versions, none of them match the java version I used on my local machine (8.0_292). Should I just leave it as the generic "java-version: '8' " ?

File: scripts/idfFailures.txt (hunk 19 above)
Comment (aksheeth-i) [reply scripts/idfFailures.txt:1]: The following error occurs in this project:

ERROR (WRITE_ERROR_STDOUT_THROWABLE): org.apache.dubbo:dubbo-registry-default:2.7.0-SNAPSHOT
java.lang.Exception: Non-zero exit code (output in /home/aksheeth_i/iDFlakies/scripts/testing-script-results/incubator- dubbo/failing-test-output-1637730318052-a56306f5-cf6b-4e4b-a8db-6bb7817cd974): 1        
         at edu.illinois.cs.testrunner.runner.Runner.$anonfun$runWithCp$3(Runner.scala:71)                                                                                                                      
         at edu.illinois.cs.testrunner.util.TempFiles$.$anonfun$withProperties$2(TempFiles.scala:38)                                                                                                            
         at edu.illinois.cs.testrunner.util.autoClose.$anonfun$autoCloseTry$1(autoClose.scala:7)
        ....

Should this be detailed or removed altogether?

Comment (aksheeth-i) [reply scripts/idfFailures.txt:1]: Will open issue request and delete idfFailures.txt

**Review by august782 — APPROVED** [review]
